app/src/main/java/org/openhdp/hdt/ui/history/ProvideHistoricEntriesUseCase.kt

```diff
@@ -0,0 +1,121 @@
+package org.openhdp.hdt.ui.history
+
+import org.openhdp.hdt.data.StopwatchRepository
+import org.openhdp.hdt.data.entities.Timestamp
+import java.text.SimpleDateFormat
+import java.util.*
+import java.util.concurrent.TimeUnit
+import javax.inject.Inject
+
+class ProvideHistoricEntriesUseCase @Inject constructor(
+    private val stopwatchRepository: StopwatchRepository
+) {
+
+    suspend fun execute(stopwatchId: String): List<DayHeader> {
+        val result = arrayListOf<DayHeader>()
+        val totalTimestamps = stopwatchRepository.timestamps(stopwatchId)
+        val timestamps = totalTimestamps.filter { it.stopTime != null }
+            .sortedBy { it.startTime }
+
+        if (timestamps.isEmpty()) {
+            return result
+        }
+
+        val firstDay = timestamps.first().toCalendarDay()
+        val lastDay = timestamps.last().toCalendarDay()
+
+        if (firstDay == lastDay) {
+            // single date?
+            includeDaysFrom(distinctDay = firstDay, timestamps, result)
+        } else {
+            // handle range of days
```

Comment:
The assumption of the timestamp is that it cannot last more than 24 hours - because when the start of the day occurs the stopwatch should automatically stop. I suggest to use the `startTime` date to determine the header date. So the day headers should be created only when there is at least one timestamp - now as I was testing I had a range of empty days between successive timestamps. To handle after midnight time (when start of the day is not 00:00) I suggest following check (pseudocode):
```
if(timestamp.first.time - startOfTheDayTime < 0) //for example 03:00:00 - 06:00:00 = -03:00:00 (just time without date) 
{
headerDay = timestamps.first().toCalendarDay() - Calendar.DAY // after midnight day, so we use the day before
}  else {
headerDay = timestamps.first().toCalendarDay() 
}
```

Reply:
will do fixes in separate PR
for now I will just skip days with 0 startTime timestamps